src/app/components/hris/employees/employee-termination/employee-termination.component.ts

```diff
@@ -0,0 +1,182 @@
+import { Component, HostListener, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { terminationOptions } from 'src/app/models/hris/constants/terminationOptions.constants';
+import { NavService } from 'src/app/services/shared-services/nav-service/nav.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SnackbarService } from 'src/app/services/shared-services/snackbar-service/snackbar.service';
+import { EmployeeService } from 'src/app/services/hris/employee/employee.service';
+import { EmployeeProfile } from 'src/app/models/hris/employee-profile.interface';
+import { EmployeeTerminationService } from 'src/app/services/hris/employee/employee-termination.service';
+
+@Component({
+  selector: 'app-employee-termination',
+  templateUrl: './employee-termination.component.html',
+  styleUrls: ['./employee-termination.component.css']
+})
+export class EmployeeTerminationComponent implements OnInit {
+
+  @Input() employeeProfile!: EmployeeProfile;
+  @HostListener('window:resize', ['$event'])
+  onResize() {
+    this.isMobileScreen = window.innerWidth < 768;
+    this.screenWidth = window.innerWidth;
+  }
+
+  newterminationform: FormGroup;
+  terminationOption: string[] = terminationOptions.map((termination) => termination.value);
+  selectedFile!: File;
+  screenWidth = window.innerWidth;
+
+  isMobileScreen: boolean = false;
+  fileUploaded: boolean = false;
+  interviewFileUploaded: boolean = false;
+  isvalidFile: boolean = false;
+  isvalidFileSize: boolean = false;
+  checkboxesValid: boolean = true;
+  formSubmitted: boolean = false;
+
+  terminationOptionValue: number = 0;
+  interviewDocFilename: string = "";
+  base64String: string = "";
+  employeeId = this.route.snapshot.params['id'];
+
+  constructor(
+    private navService: NavService,
+    private fb: FormBuilder,
+    private router: Router,
+    private snackBarService: SnackbarService,
+    public employeeService: EmployeeService,
+    private employeeTerminationService: EmployeeTerminationService,
+    private route: ActivatedRoute,
+  ) {
+    this.newterminationform = this.fb.group({
+      terminationOption: new FormControl('', Validators.required),
+    dayOfNotice: new FormControl<Date | string>(new Date(Date.now()), Validators.required),
+    lastDayOfEmployment: new FormControl<Date | string>(new Date(Date.now()), Validators.required),
+    reEmploymentStatus: new FormControl<boolean>(false, Validators.required),
+    equipmentStatus: new FormControl<boolean>(false, Validators.required),
+    accountsStatus: new FormControl<boolean>(false, Validators.required),
+    terminationDocument: new FormControl<string>('', Validators.required),
+    terminationComments: new FormControl<string>(''),
+  });
```

Comment:
Format this to keep a consistent alignment

Reply:
done